OpenSim/Auxiliary/auxiliaryTestFunctions.h

```diff
@@ -33,24 +33,61 @@
 #include <fstream>
 #include <string>
 #include <regex>
+#include <type_traits>
 
-template <typename T>
-void ASSERT_EQUAL(T expected, 
+template <typename T,
+   typename std::enable_if<std::is_floating_point<T>::value>::type* = nullptr >
+void ASSERT_EQUAL(T expected,
                   T found, 
                   T tolerance, 
                   std::string file = "", 
                   int line = -1, 
                   std::string message = "") {
+    // if both values are NaN treat them as being equivalent for the
+    // sake of comparing experimental data and results where NaNs are
+    // possible
+    if(SimTK::isNaN(found) && SimTK::isNaN(expected))
+        return;
+    if (found < expected - tolerance || found > expected + tolerance)
```

Comment:
How does NaN compare to Real? Perhaps provide a comment here, or include a case that explicitly throws if *either* `found` *or* `expected` is `NaN`.

Reply:
A NaN to real will always fail. In fact a NaN with anything fails including another NaN. That is why I have to check of both are NaN if I want the comparison to pass in that case (which we do).